providers/autodns/api.go

```diff
@@ -0,0 +1,143 @@
+package autodns
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"sort"
+
+	"github.com/StackExchange/dnscontrol/v3/models"
+)
+
+type ZoneListFilter struct {
+	Key string `json:"key"`
+	Value string `json:"value"`
+	Operator string `json:"operator"`
+	Link string `json:"link,omitempty"`
+	Filter []*ZoneListFilter `json:"filters,omitempty"`
+}
+
+type ZoneListRequest struct {
+	Filter []*ZoneListFilter `json:"filters"`
+}
+
+func (api *autoDnsProvider) request(method string, requestPath string, data interface{}) ([]byte, error) {
+	client := &http.Client{}
+
+	requestUrl := api.baseURL
+	requestUrl.Path = api.baseURL.Path + requestPath
+
+	request := &http.Request{
+		URL: &requestUrl,
+		Header: api.defaultHeaders,
+		Method: method,
+	}
+
+	if data != nil {
+		body, _ := json.Marshal(data)
+
+		buffer := bytes.NewBuffer(body)
+
+		request.Body = io.NopCloser(buffer)
+	}
+
+	response, error := client.Do(request)
```

Comment:
Style: Remove the blank lines between `Do`, the error check, and the defer. That way everything related to the Do is joined together.